### PR TITLE
Algorithm and tests

### PR DESCRIPTION
**Pull request by Andrzej97**

### CODE REVIEW

File: src/university_chatbot/university_conversation_logic_adapter.py [hunk 1]
```diff
@@ -43,30 +43,72 @@ def find_best_tags_coverage(self, documents, tags):
...
+            coverage = statement_utils.my_intersection(set(tags), set(tags_from_document))
+            confidence, was_accepted = self.is_accepted(coverage, len(set(tags_from_document)), tags)
+            if was_accepted and confidence >= max_conf_from_covered_docs:
+                id_of_best_cov_doc = document['_id']
+                max_conf_from_covered_docs = confidence
+        result_list = list(filter(lambda obj: obj['_id'] == id_of_best_cov_doc, documents))
+        if len(result_list) > 0:
+            return [result_list[0]['text'], max_conf_from_covered_docs]
+        else:
+            return [None, 0.0]
```
Comment (rpych) [inline src/university_chatbot/university_conversation_logic_adapter.py]: i think i will be cleaner to use tuples here, if it is possible? you can use tuple similar to list then

File: src/university_chatbot/university_conversation_logic_adapter.py [hunk 2]
```diff
@@ -43,30 +43,72 @@ def find_best_tags_coverage(self, documents, tags):
...
-                    result_document_lemmas, confidence_by_lemmas = find_coverage_res
-        if confidence_by_lemmas + confidence_by_tags > -2 and result_document_tags is not None:
-            if confidence_by_tags > confidence_by_lemmas:
-                res = Statement(result_document_tags)
-                res.confidence = 1
+        tags = self.create_tags(statement)
+        docs_main_collection = self.db.get_docs_from_collection_by_tags_list(Configuration.MAIN_COLLECTION.value, tags)
+        docs_phrases_collection = self.db.get_docs_from_collection_by_tags_list(Configuration.PHRASES_COLLECTION.value, tags)
+        result_main_collection = ['', 0.0]
+        result_phrases_collection = ['', 0.0]
```
Comment (rpych) [inline src/university_chatbot/university_conversation_logic_adapter.py]: tuples?

File: documents/notesAndrzej [hunk 3]
```diff
@@ -0,0 +1,7 @@
+U Andrzeja na komputerze tak się mongo odpala: /bin/mongodb/mongodb/usr/bin/mongod
+Dobry python jest tu:                          /home/andrzej/Python-3.6.9/python
+
+Statystyki funkcji w statement_utils: isMatched: (24.11.2019)
+# isMatched z elifem: elif single_tag in {'agh', 'akademia'} and tag in {'agh', 'uczelnia'}: 303/346
+# isMatched z elifem: elif single_tag in {'agh', 'akademia', 'uczelnia'} and tag in {'agh', 'akademia', 'uczelnia'}: 305/346
+# isMatched bezz elif: 300/346
```
Comment (v-tec2706) [inline documents/notesAndrzej]: hmm... I don't think we need this on the master, maybe just paste it into the wiki?

File: src/common_utils/language_utils/statement_utils.py [hunk 4]
```diff
@@ -14,7 +14,7 @@ def prepare_statement(*words):
     for word in words:
         response += word
         response += " "
-    return filter_unexpected_signs(response)
+    return response
```
Comment (v-tec2706) [inline src/common_utils/language_utils/statement_utils.py]: `filter_unexpected_signs()` is changed now, I think we should keep it

File: src/common_utils/language_utils/statement_utils.py [hunk 5]
```diff
@@ -43,4 +43,33 @@ def prepare_shortened_statement(many_sentence_response, first_index=0, length=1)
...
+
+
+def extract_single_tags(single_or_complex_tag):
+    return single_or_complex_tag.split(':')
+
+
+# isMatched z elifem: elif single_tag in {'agh', 'akademia'} and tag in {'agh', 'uczelnia'}: 303/346
+# isMatched z elifem: elif single_tag in {'agh', 'akademia', 'uczelnia'} and tag in {'agh', 'akademia', 'uczelnia'}: 305/346
+# isMatched bezz elif: 300/346
+def isMatched(single_tag, set):
```
Comment (v-tec2706) [inline src/common_utils/language_utils/statement_utils.py]: maybe a more meaningful description?

File: src/common_utils/language_utils/statement_utils.py [hunk 6]
```diff
@@ -43,4 +43,33 @@ def prepare_shortened_statement(many_sentence_response, first_index=0, length=1)
...
+    for single_or_complex_tag in set:
+        single_tags = extract_single_tags(single_or_complex_tag)
+        for tag in single_tags:
+            if single_tag == tag:
+                return True
+            elif single_tag in {'agh', 'akademia', 'uczelnia'} and tag in {'agh', 'akademia', 'uczelnia'}:
```
[the remark this answers is not in the document]
Comment (v-tec2706) [reply src/common_utils/language_utils/statement_utils.py]: or even better, change it to method argument? It will be more flexible

**Pull request merged**